### PR TITLE
feat(Date & Time Node): Overhaul of the node

### PR DESCRIPTION
**Pull request by agobrech**
Github issue / Community forum post (link here to close automatically):
https://community.n8n.io/t/date-time-node-option-to-format-is-gone-in-0-217-2/23662?u=mutedjam

### CODE REVIEW

File: packages/nodes-base/nodes/DateTime/test/node/workflow.luxonDateTime.json [hunk 1]
```diff
@@ -1,165 +0,0 @@
-{
-	"name": "My workflow 10",
-	"nodes": [
```
[the remark this answers is not in the document]
Comment (agobrech) [reply packages/nodes-base/nodes/DateTime/test/node/workflow.luxonDateTime.json:3]: I am now loading the test's setting(to get the workflow's timezone) which we didn't do before so I had to change the test. However you are right we are no longer testing the v1. I'll add a v1 test at the end of the review.

[the remark this answers is not in the document]
Comment (michael-radency) [reply packages/nodes-base/nodes/DateTime/test/node/workflow.luxonDateTime.json]: I did sync v1 with master and restored this file, test for we v1 passing

File: packages/nodes-base/nodes/DateTime/V2/AddToDateDescription.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+		default: '',
+		displayOptions: {
+			show: {
+				operation: ['addToDate'],
+			},
+		},
+	},
+	{
+		displayName: 'Date to Add To',
+		name: 'addToDate',
```
Comment (michael-radency) [inline packages/nodes-base/nodes/DateTime/V2/AddToDateDescription.ts:18]: I would change name to 'data' to preserve input if operation was changed

File: packages/nodes-base/nodes/DateTime/V2/GetTimeBetweenDates.ts [hunk 3]
```diff
@@ -81,4 +81,25 @@ export const GetTimeBetweenDatesDescription: INodeProperties[] = [
 			},
 		},
 	},
+	{
+		displayName: 'Options',
+		name: 'additionalFields',
```
Comment (michael-radency) [inline packages/nodes-base/nodes/DateTime/V2/GetTimeBetweenDates.ts:86]: why not options?

Comment (agobrech) [reply packages/nodes-base/nodes/DateTime/V2/GetTimeBetweenDates.ts]: It was an old field naming, fixed

**Review by michael-radency — APPROVED** [review]
LGTM

**Comment by janober** [comment]
Got released with [`n8n@0.228.2`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.228.2)

**Pull request merged**